common/core/desktop/src/km_kbp_options_api.cpp

```diff
@@ -25,7 +25,9 @@ km_kbp_options_list_size(km_kbp_option_item const *opts)
   if (!opts)  return 0;
 
   auto n = 0;
-  while (opts->key) ++n;
+    for (; opts->key; ++opts){
```

Comment:
Shouldn't this check for `opts != NULL` as well?

And while you're at it, the indentation is wrong for this line.

Reply:
Isn't that checked in l. 24 `assert(ops)`?

Reply:
Too much C#! I guess I meant `*opts != NULL`

Reply:
The options list should be terminated #define KM_KBP_OPTIONS_END { 0, 0, 0 }
The for loop second statement is checking if opts->key is defined and therefore the same as opts->key!=NULL.
See line 66 in the same file.

Reply:
```suggestion
  for (; opts->key; ++opts) {
```